### ISSUE TITLE
CSmartTable v4.1.3 breaks column properties

### ISSUE THREAD

**Issue by dang-cb** [coreui/coreui-vue#186]
The recent fixes to CSmartTable (thank you!) have unfortunately introduced a breaking change to column properties - they are now all ignored - only the `key` property is read.

Checking the source code, it looks like the CSmartTableHead instance has changed from
```
h$1(CSmartTableHead, {
  component: 'head',
  ...
  columns: props.columns,
  ...
},
```
to
```
h$1(CSmartTableHead, {
  component: 'head',
  ...
  columns: rawColumnNames.value,
  ...
},
```
...which I suspect is where all the extra properties are lost.

Changing that setting back appears to fix the issue, but I haven't tested it thoroughly yet to know if that has a knock on effect elsewhere...

Dan

**Comment by mrholek** [comment]
@dang-cb you're right. There should be a conditional.

`columns: props.columns ? props.columns : rawColumnNames.value,`

Fixed in v4.1.4

**Comment by dang-cb** [comment]
Oh, that's smart!

Tested and confirmed fixed in v4.1.4 - thanks once again! 🥇
